pkg/gardenlet/controller/managedseed/add.go

```diff
@@ -0,0 +1,327 @@
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+const (
+	// ControllerName is the name of this controller.
+	ControllerName = "managedseed"
+
+	// GardenletDefaultKubeconfigSecretName is the default name for the field in the Gardenlet component configuration
+	// .gardenClientConnection.KubeconfigSecret.Name
+	GardenletDefaultKubeconfigSecretName = "gardenlet-kubeconfig"
+	// GardenletDefaultKubeconfigBootstrapSecretName is the default name for the field in the Gardenlet component configuration
+	// .gardenClientConnection.BootstrapKubeconfig.Name
+	GardenletDefaultKubeconfigBootstrapSecretName = "gardenlet-kubeconfig-bootstrap"
+)
+
+// AddToManager adds Reconciler to the given manager.
+func (r *Reconciler) AddToManager(
+	mgr manager.Manager,
+	cfg config.GardenletConfiguration,
+	gardenCluster cluster.Cluster,
+	seedCluster cluster.Cluster,
+	shootClientMap clientmap.ClientMap,
+	imageVector imagevector.ImageVector,
+) error {
+	if r.GardenClient == nil {
+		r.GardenClient = gardenCluster.GetClient()
+	}
+	if r.GardenNamespaceGarden == "" {
+		r.GardenNamespaceGarden = v1beta1constants.GardenNamespace
+	}
+	if r.GardenNamespaceShoot == "" {
+		r.GardenNamespaceShoot = v1beta1constants.GardenNamespace
+	}
+	if r.ChartsPath == "" {
+		r.ChartsPath = charts.Path
+	}
+
+	valuesHelper := NewValuesHelper(&cfg, imageVector)
+
+	r.Actuator = newActuator(gardenCluster.GetConfig(),
+		gardenCluster.GetAPIReader(),
+		gardenCluster.GetClient(),
+		seedCluster.GetClient(),
+		shootClientMap,
+		valuesHelper,
+		gardenCluster.GetEventRecorderFor(ControllerName+"-controller"),
+		r.ChartsPath,
+		r.GardenNamespaceShoot,
+	)
+
+	// It's not possible to overwrite the event handler when using the controller builder. Hence, we have to build up
+	// the controller manually.
+	c, err := controller.New(
+		ControllerName,
+		mgr,
+		controller.Options{
+			Reconciler:              r,
+			MaxConcurrentReconciles: pointer.IntDeref(r.Config.ConcurrentSyncs, 0),
+			RecoverPanic:            true,
+			RateLimiter:             r.RateLimiter,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	seedName := confighelper.SeedNameFromSeedConfig(cfg.SeedConfig)
+
+	if err := c.Watch(
+		source.NewKindWithCache(&seedmanagementv1alpha1.ManagedSeed{}, gardenCluster.GetCache()),
+		r.EnqueueWithJitterDelay(r.Config),
+		r.ManagedSeedFilterPredicate(seedName),
+		&predicate.GenerationChangedPredicate{},
+	); err != nil {
+		return err
+	}
+
+	return c.Watch(
+		source.NewKindWithCache(&gardencorev1beta1.Seed{}, gardenCluster.GetCache()),
+		mapper.EnqueueRequestsFrom(mapper.MapFunc(r.MapSeedToManagedSeed), mapper.UpdateWithNew, c.GetLogger()),
+		r.SeedOfManagedSeedFilterPredicate(seedName),
+	)
+}
+
+// ManagedSeedFilterPredicate returns the predicate for ManagedSeed events.
+func (r *Reconciler) ManagedSeedFilterPredicate(seedName string) predicate.Predicate {
```

Comment:
```suggestion
func (r *Reconciler) ManagedSeedPredicate() predicate.Predicate {
```

fetch the name from the config like above